src/coreclr/src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -484,54 +484,23 @@ private static int CalculateOptimalMaxSpinWaitsPerSpinIteration()
         }
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern int GetCurrentProcessorNumber();
+        internal static extern int GetCurrentProcessorNumber();
 
-        // The upper bits of t_currentProcessorIdCache are the currentProcessorId. The lower bits of
-        // the t_currentProcessorIdCache are counting down to get it periodically refreshed.
-        // TODO: Consider flushing the currentProcessorIdCache on Wait operations or similar
-        // actions that are likely to result in changing the executing core
-        [ThreadStatic]
-        private static int t_currentProcessorIdCache;
-
-        private const int ProcessorIdCacheShift = 16;
-        private const int ProcessorIdCacheCountDownMask = (1 << ProcessorIdCacheShift) - 1;
-        private const int ProcessorIdRefreshRate = 5000;
-
-        private static int RefreshCurrentProcessorId()
-        {
-            int currentProcessorId = GetCurrentProcessorNumber();
-
-            // On Unix, GetCurrentProcessorNumber() is implemented in terms of sched_getcpu, which
-            // doesn't exist on all platforms.  On those it doesn't exist on, GetCurrentProcessorNumber()
-            // returns -1.  As a fallback in that case and to spread the threads across the buckets
-            // by default, we use the current managed thread ID as a proxy.
-            if (currentProcessorId < 0) currentProcessorId = Environment.CurrentManagedThreadId;
-
-            // Add offset to make it clear that it is not guaranteed to be 0-based processor number
-            currentProcessorId += 100;
```

Comment:
It looks like the offset was removed in this cache refactoring PR. Are we going to guarantee a 0-based processor number or not? Whichever we go with, I think the docs ought to be updated to reflect that and I'll happily PR it. I think if we're not going to guarantee it though, the offset is a good idea and should be reintroduced. @VSadov @jkotas

Reply:
We do not guarantee 0-based processor number.

Whether or not to pay the extra cycle to add the offset is an interesting question. I do not have a strong opinion either way.

Reply:
@CoffeeFlux - We cannot guarantee the `[0..CpuCount)` range. The underlying API may be nonfunctional and managed threadID used instead. On VMs the core ID could be outside of this range too.

Why do you think adding offset is a good idea though?. 
It adds some cost to the API while anyone doing something like  `GetCurrentProcessorId & mask` will be tempted to subtract the offset.

Reply:
My fear is customers relying on the behavior anyway and then ending up with Mono bug reports because we can't guarantee it on mobile, but maybe we should just update the docs to be unambiguous about this. My opinion isn't that strong, I just want it to be clarified somewhere other than comments in the source code.

Reply:
After sleeping on it, I think I'm okay just leaving the value as-is and updating the docs. I'll PR the change later today.